src/core/pointcloud/qgspointclouddataprovider.h

```diff
@@ -279,10 +279,10 @@ class CORE_EXPORT QgsPointCloudDataProvider: public QgsDataProvider
     QString mSubsetString;
 
     //! Identify in a specific index (used for sub-indexes)
-    QVector<QVariantMap> identify( QgsPointCloudIndex *index, double maxError, const QgsGeometry &extentGeometry, const QgsDoubleRange &extentZRange, int pointsLimit ) SIP_SKIP ;
+    QVector<QVariantMap> identify( QgsPointCloudIndex index, double maxError, const QgsGeometry &extentGeometry, const QgsDoubleRange &extentZRange, int pointsLimit ) SIP_SKIP ;
```

Comment:
```suggestion
    QVector<QVariantMap> identify( const QgsPointCloudIndex &index, double maxError, const QgsGeometry &extentGeometry, const QgsDoubleRange &extentZRange, int pointsLimit ) SIP_SKIP ;
```
elsewhere too

Reply:
Thanks for the comments, I'll look through them tomorrow. But I don't think `QgsPointCloudIndex` should be passed by reference. The point of the class is that it's basically an `std::shared_ptr`. Passing it by reference might remove a refcount increment at the cost of more indirection, but I'd rather just pass it by value for simplicity.

Reply:
If we had the full range of C++ to play with, I'd personally either use `std::shared_ptr<QgsAbstractPointCloudIndex>` or `QgsAbstractPointCloudIndex &`, but as-is I'd just use the wrapper everywhere.

Reply:
Hmm -- I would still push towards references, in all cases EXCEPT those where we know that we're explicitly storing the result in a member variable (and could possibly benefit from a move).

(In general we try to follow Qt's approach for things like this, and they **always** pass QString by reference, even though it's implicitly shared and copies are cheap.)